### PR TITLE
Move IAddMissingImportsFeatureService to use OOP

### PR DESCRIPTION
**Pull request by ryzngard**
IAddMissingImportsFeatureService wraps IAddImportFeatureService with some logic for reducing fixes that will be taken. The work to get diagnostics was previously done inproc, which can be expensive. This moves that work to IAddImportFeatureService via a new `GetUniqueFixesAsync` api and utilizes OOP when possible

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
I'll try to get to this this weekend. If I can't, long me on Monday!  Thanks!

**Comment by ryzngard** [comment]
> I'll try to get to this this weekend. If I can't, long me on Monday! Thanks!

This can definitely wait until Monday @CyrusNajmabadi

**Comment by ryzngard** [comment]
> I'll try to get to this this weekend. If I can't, long me on Monday! Thanks!

This can definitely wait until Monday @CyrusNajmabadi

**Comment by ryzngard** [comment]
ping @JoeRobich @CyrusNajmabadi if you have some time

File: src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsFeatureService.cs [hunk 1]
```diff
@@ -52,15 +51,17 @@ public async Task<Document> AddMissingImportsAsync(Document document, AddMissing
         public async Task<AddMissingImportsAnalysisResult> AnalyzeAsync(Document document, TextSpan textSpan, CancellationToken cancellationToken)
         {
             // Get the diagnostics that indicate a missing import.
```
Comment (JoeRobich) [inline src/Features/Core/Portable/CodeRefactorings/AddMissingImports/AbstractAddMissingImportsFeatureService.cs:53]: Stale comment
```suggestion
```

File: src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs [hunk 2]
```diff
@@ -517,6 +517,74 @@ private static bool NotNull(SymbolReference reference)
...
+            bool searchReferenceAssemblies,
+            ImmutableArray<PackageSource> packageSources,
+            CancellationToken cancellationToken)
+        {
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var placeSystemNamespaceFirst = documentOptions.GetOption(GenerationOptions.PlaceSystemNamespaceFirst);
+            var allowInHiddenRegions = document.CanAddImportsInHiddenRegions();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var diagnostics = semanticModel.GetDiagnostics(span, cancellationToken)
```
Comment (JoeRobich) [inline src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs:558]: Moved comment from below
```suggestion

            // Get the diagnostics that indicate a missing import.
            var diagnostics = semanticModel.GetDiagnostics(span, cancellationToken)
```

**Review by JoeRobich — APPROVED** [review]

**Pull request merged**